apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-title/edit.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import React, { Component } from 'react';
+import { __ } from '@wordpress/i18n';
+const { withSelect } = wp.data;
+const { apiFetch } = wp;
+
+/**
+ * Internal dependencies
+ */
+
+class SiteTitleEdit extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			title: null,
+			initialTitle: null,
+		};
+	}
+
+	componentDidMount() {
+		if ( this.state.initialTitle ) {
+			return;
+		}
+
+		apiFetch( { path: '/wp/v2/settings' } ).then( ( { title } ) => {
+			this.setState( { initialTitle: title } );
+		} );
```

Comment:
Hmm, I do wonder how often we should query. This is probably okay for now, but we might want to follow up on this. 

Data here may get stale in a long editing session. We previously had problems with state, in Gutenlypso when switching sites, but it isn't a problem in this implementation.

For context in Calypso, we normally use a data pattern of a query component. example: https://github.com/Automattic/wp-calypso/blob/master/client/components/data/query-comment/index.jsx 

The idea is make a request on mount, and any time related props change like site id.